packages/core/admin/admin/src/components/MainNav/NavUser.tsx

```diff
@@ -34,30 +52,39 @@ const MenuItem = styled(Menu.Item)`
     display: flex;
     align-items: center;
     gap: ${({ theme }) => theme.spaces[3]};
-    justify-content: space-between;
   }
 `;
 
 const MenuItemDanger = styled(MenuItem)`
   &:hover {
-    ${({ theme }) => `
-    background: ${theme.colors.danger100};
-  `}
+    background: ${({ theme }) => theme.colors.danger100};
   }
 `;
 
-export const NavUser = ({ children, initials, ...props }: NavUserProps) => {
+export interface NavUserProps extends ButtonProps {
+  initials: string;
+}
+
+export const NavUser = ({ initials, ...props }: NavUserProps) => {
   const { formatMessage } = useIntl();
   const navigate = useNavigate();
+  const user = useAuth('User', (state) => state.user);
   const logout = useAuth('Logout', (state) => state.logout);
+
   const handleProfile = () => {
     navigate('/me');
   };
+
   const handleLogout = () => {
     logout();
     navigate('/auth/login');
   };
 
+  const displayName =
+    user?.firstname && user?.lastname
+      ? `${user.firstname} ${user.lastname}`
+      : user?.username || 'Unknown User';
```

Comment:
We have a `getDisplayName` util in: packages/core/admin/admin/src/utils/users.ts Could you use that here?